Master does not compile with 0.16.0

It seem that this depends on tree-sitter-c which depends on a version of tree-sitter that does not compile with zig 0.16.0.

Its the same error that the CI has: [CI log](https://github.com/tree-sitter/zig-tree-sitter/actions/runs/24640286719/job/72042913398#step:6:21)


```sh
~/zig-tree-sitter/ > nix develop . -c zig build
zig-pkg/tree_sitter-0.25.0-8heIf4xwAQCcM_rEJFSvWcc0Y6TaXQPw1kFZRWoea4Ne/build.zig:23:8: error: no field or member function named 'linkLibrary' in 'Build.Step.Compile'
    lib.linkLibrary(core.artifact("tree-sitter"));
    ~~~^~~~~~~~~~~~
/nix/store/9zyky24fffkbhnkmjcnrjcl3dmvx6w9x-zig-0.16.0/lib/std/Build/Step/Compile.zig:1:1: note: struct declared here
const Compile = @This();
^~~~~
zig-pkg/tree_sitter-0.25.0-8heIf4xwAQCcM_rEJFSvWcc0Y6TaXQPw1kFZRWoea4Ne/build.zig:23:8: note: method invocation only supports up to one level of implicit pointer dereferencing
zig-pkg/tree_sitter-0.25.0-8heIf4xwAQCcM_rEJFSvWcc0Y6TaXQPw1kFZRWoea4Ne/build.zig:23:8: note: use '.*' to dereference pointer
referenced by:
    runBuild__anon_141352: /nix/store/9zyky24fffkbhnkmjcnrjcl3dmvx6w9x-zig-0.16.0/lib/std/Build.zig:2265:44
    dependencyInner__anon_138706: /nix/store/9zyky24fffkbhnkmjcnrjcl3dmvx6w9x-zig-0.16.0/lib/std/Build.zig:2245:29
    9 reference(s) hidden; use '-freference-trace=11' to see all references
zig-pkg/tree_sitter-0.25.9-Tw2sR_nDCwDWImicW-dekLrHo1GO04pj2MiTtTBBxjlK/build.zig:23:12: error: no field or member function named 'addCSourceFile' in 'Build.Step.Compile'
        lib.addCSourceFile(.{
        ~~~^~~~~~~~~~~~~~~
/nix/store/9zyky24fffkbhnkmjcnrjcl3dmvx6w9x-zig-0.16.0/lib/std/Build/Step/Compile.zig:1:1: note: struct declared here
const Compile = @This();
^~~~~
zig-pkg/tree_sitter-0.25.9-Tw2sR_nDCwDWImicW-dekLrHo1GO04pj2MiTtTBBxjlK/build.zig:23:12: note: method invocation only supports up to one level of implicit pointer dereferencing
zig-pkg/tree_sitter-0.25.9-Tw2sR_nDCwDWImicW-dekLrHo1GO04pj2MiTtTBBxjlK/build.zig:23:12: note: use '.*' to dereference pointer
zig-pkg/tree_sitter_c-0.24.1-ns1YLwdjQACIi5BEwewiqu6a6Zcl91RWoSWbz0KmxUOj/build.zig:23:8: error: no field or member function named 'addCSourceFile' in 'Build.Step.Compile'
    lib.addCSourceFile(.{
    ~~~^~~~~~~~~~~~~~~
/nix/store/9zyky24fffkbhnkmjcnrjcl3dmvx6w9x-zig-0.16.0/lib/std/Build/Step/Compile.zig:1:1: note: struct declared here
const Compile = @This();
^~~~~
zig-pkg/tree_sitter_c-0.24.1-ns1YLwdjQACIi5BEwewiqu6a6Zcl91RWoSWbz0KmxUOj/build.zig:23:8: note: method invocation only supports up to one level of implicit pointer dereferencing
zig-pkg/tree_sitter_c-0.24.1-ns1YLwdjQACIi5BEwewiqu6a6Zcl91RWoSWbz0KmxUOj/build.zig:23:8: note: use '.*' to dereference pointer

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -30,7 +30,8 @@ jobs:
       matrix:
         target:
           - x86_64-linux-gnu
-          - x86_64-linux-musl
+          # Zig 0.16 needs musl-1.2.5 (in ubuntu-26.04)
+          # - x86_64-linux-musl
           - aarch64-linux-gnu
           - x86_64-windows-gnu
           - x86_64-windows-msvc
@@ -42,8 +43,9 @@ jobs:
         include:
           - os: ubuntu-24.04
             target: x86_64-linux-gnu
-          - os: ubuntu-24.04
-            target: x86_64-linux-musl
+          # Zig 0.16 needs musl-1.2.5 (in ubuntu-26.04)
+          # - os: ubuntu-24.04
+          #   target: x86_64-linux-musl
           - os: ubuntu-24.04-arm
             target: aarch64-linux-gnu
           - os: windows-2022

--- a/build.zig.zon
+++ b/build.zig.zon
@@ -5,8 +5,8 @@
     .minimum_zig_version = "0.16.0",
     .dependencies = .{
         .@"tree-sitter-c" = .{
-            .url = "git+https://github.com/tree-sitter/tree-sitter-c#586dd62ad4d3532895ea80a8382d21307b4733bd",
-            .hash = "tree_sitter_c-0.24.1-ns1YLwdjQACIi5BEwewiqu6a6Zcl91RWoSWbz0KmxUOj",
+            .url = "git+https://github.com/tree-sitter/tree-sitter-c#b780e47fc780ddc8da13afa35a3f4ed5c157823d",
+            .hash = "tree_sitter_c-0.24.2-y5boS-ptQADHoCoVfjGT_nFtFQ5LbomIkW0fxG3_cmdB",
             .lazy = true,
         },
         .tree_sitter = .{

